### PR TITLE
Cleanup LaneMinion

### LINKED ISSUE #969: LaneMinion Mess
This class has many functions which loop back on themselves due to Update(), which calls Minion functions, which are the same as LaneMinion functions. This also means LaneMinion hides Minion functions.
Overall LaneMinion is horribly unoptimized and creates a lot of performance issues.

### PR DESCRIPTION
**Pull request by lzardy**
Removes loop-back issues and removes functions which hide parent functions.
Gives a big boost to performance even though this is such a small change.
If there are any other concerns when looking at Minion.cs or LaneMinion.cs, please address them at your leisure as these parts of LS will likely be ignored for quite a while, while bigger, more baseline issues are addressed.
Resolves #969

### CODE REVIEW

File: GameServerLib/GameObjects/AttackableUnits/AI/Minion.cs [hunk 1]
```diff
@@ -103,6 +103,10 @@ public override void Update(float diff)
                         KeepFocusingTarget(); // attack/follow target
                     }
                 }
+                else if (this is ILaneMinion lm)
```
Comment (danil179) [inline GameServerLib/GameObjects/AttackableUnits/AI/Minion.cs]: It doesn't look clean to me. Minion should have no knowledge about LaneMinion (I think this implicated from Liskov substitution principle). In this case you want a function like `AIMove` that will do the walk to destination in LaneMinion but will do other things in Minion class.

I think that you can put:
```
if (ScanForTargets()) // returns true if we have a target
                {
                    if (!RecalculateAttackPosition())
                    {
                        KeepFocusingTarget(); // attack/follow target
                    }
                }
```
in a function (call it `AIMove` or something like that in Minion class, and then override it in LaneMinion to be:
```
base.AIMove()
lm.WalkToDestination(); // walk to destination (or target)
```

**Review by danil179 — CHANGES_REQUESTED** [review]
More cleaning needed.

File: GameServerLib/GameObjects/AttackableUnits/AI/LaneMinion.cs [hunk 2]
```diff
@@ -25,6 +25,7 @@ public class LaneMinion : Minion, ILaneMinion
             uint netId = 0
         ) : base(game, null, 0, 0, "", "", 1100, netId)
         {
+            IsLaneMinion = true;
```
Comment (MatthewFrench) [inline GameServerLib/GameObjects/AttackableUnits/AI/LaneMinion.cs:28]: In a future ticket, we could probably turn this into a base function that returns a static true or false and have LaneMinion override the function to return true. Then we wouldn't need a variable.

**Review by MatthewFrench — APPROVED** [review]

**Review by danil179 — APPROVED** [review]
Approved. You need to rebase in order to rebuild (change the commit message).

**Pull request merged**